Jenkinsfile - ImageStream import issue and builds failures

jcpowermac opened the pull request:
- Adds build checking and error handling if failures occur.
- Adds checking for ImageStreamTag import failure - which has
been occurring at the first build.
- Updates the application OpenShift template.  Change `PROJECT` to
`REGISTRY_PROJECT` which better represents the parameter.

codificat commented on Jenkinsfile:
```diff
@@ -24,6 +25,7 @@ pipeline {
                 syncOpenShiftSecret params.STAGE_SECRET_NAME
             }
         }
+        /*
```
Why comment out the testing stage? I guess this is to speed up testing? :wink:

jcpowermac replied:
@codificat yup especially since I have to run a "few" times to get it working right.

codificat commented on Jenkinsfile:
```diff
@@ -54,6 +57,7 @@ pipeline {
                                             "-p",
                                             "TAG=${env.TAG}",
                                             "IMAGESTREAM_TAG=${params.IMAGE_STREAM_LATEST_TAG}",
+                                            "REGISTRY_PROJECT=${params.REGISTRY_PROJECT}",
```
Good addition! I was working on something similar but yours looks much better. I'd only suggest documenting the role of `REGISTRY_PROJECT` a bit more

codificat commented on Jenkinsfile:
```diff
@@ -84,11 +88,39 @@ pipeline {
+                                }
+                            }
+                            if(currentBuild.result == 'FAILURE') {
+                                error("some failure")
```
We probably can't point to the exact cause of the failure, but maybe the error message can be a bit more informative? like point to the build logs...

jcpowermac replied:
@codificat yeah I can print the logs.  Will add that.

codificat commented on app/openshift/nodejs-mongodb-persistent.json:
```diff
@@ -460,6 +460,13 @@
     },
     {
+      "name": "REGISTRY_PROJECT",
+      "displayName": "Registry project",
```
In line of documenting this var a bit better, I'd suggest a more explicit display name... like "namespace in the registry where images are shared across environments" or something like that.

It might also be worth adding a comment in `host_vars/registry.example` around `project_name`

jcpowermac replied:
@codificat will do.

codificat reviewed (changes requested):
Some minor comments - generally looks great. I tested the changes and they work, better than before.

Requesting changes mainly about squashing and the commented-out test stage; the improvements in registry namespace handling can go in a separate PR

jcpowermac committed:
Jenkinsfile - ImageStream import issue and builds failures

- Adds build checking and error handling if failures occur.
- Adds checking for ImageStreamTag import failure - which has
been occurring at the first build.
- Updates the application OpenShift template.  Change `PROJECT` to
`REGISTRY_PROJECT` which better represents the parameter.

codificat reviewed (approved):
Tested the latest changes successfully.
/LGTM
